readme.md

```diff
@@ -59,6 +59,24 @@ and executes logic from the `protocols` package. Data required for the secure cr
 More detailed information can be found in each package's respective _readme_.
 ![architecture](./client/architecture.png)
 
+## On-chain code
+
+`go-nitro` was originally being developed "off-chain first". Alongside that development, we are now building out the on-chain component (solidity contracts): these are housed in the `nitro-protocol` directory. This directory contains an npm package with a hardhat / typechain / jest toolchain. This code originates from the nitro-protocol [npm package](https://www.npmjs.com/package/@statechannels/nitro-protocol), but is being modified / upgraded. The new version of the code is sometimes called nitro v2 or "satp".
+
+Initially, these two development streams will remain independent, with each having its own continuous integration checks.
+
+Over time, the development streams will become more coupled.
+
+We now use `solc` and `abigen` to generate Go bindings from our `*.sol` source files. This is achieved by running the `generate_adjudicator_bindings.sh` script at the top level of the repo. Because our `*.sol` files depend on external projects via `node_modules`, to run this script successfully you must:
+
+- have successfully run `npm install` in the `nitro-protocol` directory.
```

Comment:
Any particular reason we are using `npm` instead of `yarn`?

Reply:
When I try to run `npm install` in nitro-protocol directory I see:
```
➜  nitro-protocol git:(go-solidity-signature-test-rewrite) ✗ npm install
npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/@statechannels/devtools/node_modules/ganache-core/node_modules/web3-provider-engine/node_modules/eth-sig-util/ethereumjs-abi@git+https:/github.com/ethereumjs/ethereumjs-abi.git" as it does not contain a package.json file.
```

Any ideas on how to resolve that?

Reply:
@kerzhner , I had the similar issue, but after removing `node_modules` folder and running `npm install` again it was resolved.

Reply:
Got it! On my machine, there is no `node_modules` folder to remove.